l2geth/core/state_processor.go

```diff
@@ -105,6 +105,11 @@ func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *commo
 	context := NewEVMContext(msg, header, bc, author)
 	if vm.UsingOVM {
 		context.BlockNumber = msg.L1BlockNumber()
+		if msg.L1MessageSender() == nil {
+			context.L1MessageSender = common.HexToAddress("0x00000000000000000000000000000000000beef")
```

Comment:
This `vm.UsingOVM` isn't self documenting, it should have a comment describing why it exists so that it is very obvious to newcomers to the codebase what the diff is doing. We should also make `0x..beef` declared at the top so that we don't need to allocate a new one each time

Reply:
This is also a bit of a leaky abstraction, I feel like ideally we assign the things to the context inside of `NewEVMContext`. I know that this wasn't done originally as there was time pressure and we weren't as familiar with the entire codebase. This will ensure that all places that create a new evm context will get the variables assigned correctly